src/corehost/cli/setup.cmake

```diff
@@ -60,6 +60,8 @@ endif()
 # containing the reference instead of using definitions from other modules.
 if(${CMAKE_SYSTEM_NAME} MATCHES "Linux")
     set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -Xlinker -Bsymbolic -Bsymbolic-functions")
+    set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -Wl,--build-id=sha1")
+    set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -Wl,--build-id=sha1")
     add_compile_options(-fstack-protector-strong)
```

Comment:
@ellismg LGTM. For my education, the BuildID[SHA1] will be stored somewhere for us to know which one it is?

Actually, if you crack open the ELF binary and look at the strings, search for "commit" (if you use :%!xxd in vim), you can get the SHA1 etc.